x-pack/packages/kbn-elastic-assistant/impl/knowledge_base/knowledge_base_settings_management/document_entry_editor.tsx

```diff
@@ -0,0 +1,135 @@
+}
+
+export const DocumentEntryEditor: React.FC<Props> = React.memo(({ entry, setEntry }) => {
+  // Name
+  const setName = useCallback(
+    (e) => setEntry((prevEntry) => ({ ...prevEntry, name: e.target.value })),
+    [setEntry]
+  );
+
+  // Sharing
+  const setSharingOptions = useCallback(
+    (value) =>
+      setEntry((prevEntry) => ({
+        ...prevEntry,
+        users: value === i18n.SHARING_GLOBAL_OPTION_LABEL ? [] : undefined,
+      })),
+    [setEntry]
+  );
+  // TODO: KB-RBAC Disable global option if no RBAC
+  const sharingOptions = [
+    {
+      value: i18n.SHARING_PRIVATE_OPTION_LABEL,
+      inputDisplay: (
+        <EuiText size={'s'}>
+          <EuiIcon
+            color="subdued"
+            style={{ lineHeight: 'inherit', marginRight: '4px' }}
+            type="lock"
+          />
+          {i18n.SHARING_PRIVATE_OPTION_LABEL}
+        </EuiText>
+      ),
+    },
+    {
+      value: i18n.SHARING_GLOBAL_OPTION_LABEL,
+      inputDisplay: (
+        <EuiText size={'s'}>
+          <EuiIcon
+            color="subdued"
+            style={{ lineHeight: 'inherit', marginRight: '4px' }}
+            type="globe"
+          />
+          {i18n.SHARING_GLOBAL_OPTION_LABEL}
+        </EuiText>
+      ),
+    },
+  ];
+  const selectedSharingOption =
+    entry?.users?.length === 0 ? sharingOptions[1].value : sharingOptions[0].value;
+
+  // Text / markdown
+  const setMarkdownValue = useCallback(
+    (value: string) => {
+      setEntry((prevEntry) => ({ ...prevEntry, text: value }));
+    },
+    [setEntry]
+  );
+
+  // Required checkbox
+  const onRequiredKnowledgeChanged = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setEntry((prevEntry) => ({ ...prevEntry, required: e.target.checked }));
+    },
+    [setEntry]
+  );
+
+  return (
+    <EuiForm>
+      <EuiFormRow label={i18n.ENTRY_NAME_INPUT_LABEL} fullWidth>
+        <EuiFieldText
+          name="name"
+          placeholder={i18n.ENTRY_NAME_INPUT_PLACEHOLDER}
+          fullWidth
+          value={entry?.name}
+          onChange={setName}
+        />
+      </EuiFormRow>
+      <EuiFormRow
+        label={i18n.ENTRY_SHARING_INPUT_LABEL}
+        helpText={i18n.SHARING_HELP_TEXT}
+        fullWidth
+      >
+        <EuiSuperSelect
+          options={sharingOptions}
+          valueOfSelected={selectedSharingOption}
+          onChange={setSharingOptions}
+          fullWidth
+        />
+      </EuiFormRow>
+      <EuiFormRow label={i18n.ENTRY_MARKDOWN_INPUT_TEXT} fullWidth>
+        <EuiMarkdownEditor
+          aria-label={i18n.ENTRY_MARKDOWN_INPUT_TEXT}
+          placeholder="# Title"
+          value={entry?.text ?? ''}
+          onChange={setMarkdownValue}
+          height={400}
+          initialViewMode={'editing'}
+        />
+      </EuiFormRow>
+      <EuiFormRow fullWidth helpText={i18n.ENTRY_REQUIRED_KNOWLEDGE_HELP_TEXT}>
+        <EuiCheckbox
+          label={i18n.ENTRY_REQUIRED_KNOWLEDGE_CHECKBOX_LABEL}
```

Comment:
I'm wondering if the expected behavior of the `Required knowledge` option, which is described in the UI as:

`Check to indicate a Knowledge Base entry that's included in every conversation`

is consistent with the observations in the scenario below.

### Scenario

This scenario has three KB entries, all for the same user + space:

![scenario](https://github.com/user-attachments/assets/0ce136ac-277a-4873-8871-69d7701cc892)

![my_favorite_color](https://github.com/user-attachments/assets/9f1c1786-34e9-4f01-80f0-9a4e746acebc)

![my_favorite_season](https://github.com/user-attachments/assets/4b67d23a-570c-4f0e-9962-f026a241b7d3)

![my_favorite_car](https://github.com/user-attachments/assets/cd25dd1f-c4d0-46ab-8dd6-3903257fe535)

In the scenario above:

- Only the "my favorite car" entry has the `Required knowledge` checkbox checked

When the assistant is asked:

> What is my favorite color?

The assistant replied with the expected result:
![what_is_my_favorite_color](https://github.com/user-attachments/assets/1075aefb-ba89-4be8-b1b5-8b7c0b322a78)

The LangSmith output for this question was:

![my_favorite_color_langsmith](https://github.com/user-attachments/assets/86f4537d-f191-41b9-8721-a9bf077528db)

**Observations**

- All three KB entries are included as context

In a new conversation, when the assistant is asked:

> Do I have any open alerts?

The assistant correctly answers: 

![do_i_have_any_open_alerts](https://github.com/user-attachments/assets/ea8e07d3-1dca-44f0-ad92-7246958b0603)

The LangSmith output for this question was:

![my_favorite_color_langsmith](https://github.com/user-attachments/assets/76a4cb60-ad7c-4038-acc7-b869f7bfad6f)

**Observations**

- None of the three KB entries appear to be included as context
  - Specifically, the `Required knowledge` entry (`my favorite car`) is NOT included

Is the "my favorite car" KB entry expected to be included in the context sent to the assistant when asked `Do I have any open alerts?`, because it's marked as `Required knowledge`, even though a different tool is required to answer the question?

Reply:
Not just yet as the prompt still needs to be updated to include the `{{knowledge_history}}` template and the graph to do the initial retrieval of all `required` entries. My apologies for not having this in the description -- I will update the `TODO` section to include this.

This was originally plumbed in the initial move to LangGraph [here](https://github.com/spong/kibana/blob/f9505d95bb3ae8efea261336df42f501c62796db/x-pack/plugins/elastic_assistant/server/lib/langchain/graphs/default_assistant_graph/prompts.ts#L11-L13) (look for `{{knowledge_history}}` in both openAI and structuredChat prompts), but was removed before release since the v2 KB was not ready.

I will work this as part of the other prompt update you had mentioned above ('only answer elastic security questions...") since both will require plumbing the FF down to the prompt.